### PR TITLE
Add OpenSearch container

### PR DESCRIPTION
**Pull request by maltehedderich**
You have implemented a new container and would like to contribute it? Great! Here are the necessary steps:

- [x] You have added the new container as a module in the `testcontainers` directory (such as `testcontainers/my_fancy_container.py`).
- [x] You have added any new python dependencies in the `extras_require` section of `setup.py`.
- [x] You have added the `extra_requires` key to `requirements.in`.
- [x] You have updated all python requirements by running `make requirements` from the root directory.
- [x] You have added tests for the new container in the `tests` directory, e.g. `tests/test_my_fancy_container.py`.
- [x] You have added the name of the container (such as `my_fancy_container`) to the `test-components` matrix in `.github/workflows/main.yml` to ensure the tests are run.
- [x] You have rebased your development branch on `master` (or merged `master` into your development branch).

### CODE REVIEW

**Comment by maltehedderich** [comment]
Flaky Test? I can't see any relation to my changes.

File: docs/index.rst [hunk 1]
```diff
@@ -23,4 +23,5 @@ Usage modes
    RabbitMQ container <rabbitmq>
    Redis container <redis>
    Minio container <minio>
+   OpeanSearch container <opensearch>
```
Comment (tillahoffmann) [inline docs/index.rst:26]:
```suggestion
   OpenSearch container <opensearch>
```

File: docs/opensearch.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+OpeanSearch
```
Comment (tillahoffmann) [inline docs/opensearch.rst:1]:
```suggestion
OpenSearch
```

File: docs/opensearch.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+OpenSearch
+===================
+
+Allows to spin up OpeanSearch Container.
```
Comment (tillahoffmann) [inline docs/opensearch.rst:4]:
```suggestion
Allows to spin up OpenSearch Container.
```

File: README.rst [hunk 4]
```diff
@@ -12,22 +12,28 @@ Python port for testcontainers-java that allows using docker containers for func
...
+* Keycloak container
+* LocalStack container
 * MariaDb container
+* Minio container
+* MongoDB container
+* Microsoft SQL Server container
+* MySql Db container
 * Neo4j container
+* NGINX container
+* OpeanSearch container
```
Comment (tillahoffmann) [inline README.rst:30]:
```suggestion
* OpenSearch container
```

**Comment by maltehedderich** [comment]
Thanks!

File: testcontainers/opensearch.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        security_disabled=True,
+        **kwargs,
+    ):
+        """
+        Args:
+            image (str, optional): The Docker image to use for the container.
+                                   Defaults to "opensearchproject/opensearch:2.4.0".
+            port_to_expose (int, optional): The port to expose on the container.
+                                            Defaults to 9200.
+            security_disabled (bool, optional): `True` disables the security plugin in OpenSearch.
```
Comment (tillahoffmann) [inline testcontainers/opensearch.py:44]: What do you think about changing this to `security_enabled` to avoid any confusion about double negatives?

Comment (maltehedderich) [reply testcontainers/opensearch.py]: No strong opinion on this, can live with both, tweaked it.

File: testcontainers/opensearch.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            security_enabled (bool, optional): `False` disables the security plugin in OpenSearch.
+                                                Defaults to False.
+        """
+        super(OpenSearchContainer, self).__init__(image, **kwargs)
+        self.port_to_expose = port_to_expose
+        self.security_enabled = security_enabled
+
+        self.with_exposed_ports(self.port_to_expose)
+        self.with_env("discovery.type", "single-node")
+        self.with_env("plugins.security.disabled", f"{'false' if security_enabled else 'true'}")
```
Comment (tillahoffmann) [inline testcontainers/opensearch.py:53]:
```suggestion
        self.with_env("plugins.security.disabled", "false" if security_enabled else "true")
```

**Review by tillahoffmann — APPROVED** [review]
Great, thank you!

**Pull request merged**